Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -508,6 +512,14 @@ void ImageEditor::keyup_event(GUI::KeyEvent& event)
     if (!m_active_tool->is_overriding_alt() && event.key() == Key_LeftAlt)
         update_tool_cursor();
 
+    if (event.key() == Key_Space) {
+        stop_panning();
+
+        // The set_override_cursor is called in stop_panning, but for some reason it doesn't really reset the cursor
+        // But if it's being called here, it does reset the cursor. Weird...
+        set_override_cursor(m_active_cursor);
```

Comment:
As discussed on Discord, it would be great if for this revision we could at least move the bandaid fix into `AbstractZoomPanWidget` (in the form of `if (!m_is_panning)`) and with a comment noting that we are doing this because keydown events get duplicated on key hold.